### DISCUSSION TITLE
Volatile declared entities

### DISCUSSION THREAD

**Discussion by bennydiamond** [taligentx/dscKeybusInterface#68] [dscKeybusInterface Discussion]
I fail to see the relevance in declaring certain variables and function parameters as volatile in the library. At least in release code.

Removing all volatile specifiers throughout the library's code yeilded no apparent ill-effect but reduced DATA memory usage by 20 bytes and PROGRAM by 150 bytes, on ATMEGA328p at least. It probably also speeds up overall execution time a bit since compiler can freely optimize them.

So am I missing something? I have a fairly basic Alarm System setup so maybe I am not getting the full picture here.

**Comment by taligentx** [comment]
Hi @bennydiamond - `volatile` is required for variables that are modified during an interrupt - see "About Interrupt Service Routines":

https://www.arduino.cc/reference/en/language/functions/external-interrupts/attachinterrupt/

I tested this during development and confirmed that CRC errors would crop up in the data.
